core/templates/dev/head/components/share/sharing_collection_links_directive.html

```diff
@@ -0,0 +1,74 @@
+<script type="text/ng-template" id="components/sharingCollectionLinks">
```

Comment:
yes, that's an alternate approach i was talking about in a comment above. It would involve a lot of logic statements to be added to the SharingLinksDirective.js and sharing-links-directive.html.

I chose creating separate files as collection_player and exploration_player are different components and their sharing directives may belong to them individually. 

If you say, we can choose generalizing and I'll think of all changes required and try to do them in SharingLinksDirective.js and sharing_links_directive.html instead.